frontend/components/Homepage.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from 'react';
+import Link from 'react-router/lib/Link';
+import browserHistory from 'react-router/lib/browserHistory';
+import Loading from './Loading.jsx';
+
+import rootUrl from '../endpoint.js';
+
+export default class Homepage extends Component {
+
+    constructor(opts) {
+        super(opts);
+
+        this.state = {
+            options: []
+        }
+
+        this.submitForm = this.submitForm.bind(this);
+    }
+
+    submitForm(ev) {
+
+        ev.preventDefault();
+
+        if (!this.refs.name.value) { alert("Please provide a valid name."); }
```

Comment:
this should add the .error class to the field and return false so it doesn't continue, instead of triggering an alert. other small things: the error class should remove on keyup, so when they start typing again it looks valid. we should also not allow non numbers and , into the amount field:

```
  $(document).on("keyup", "input.error", function() {
    $(this).removeClass("error");
  });

  $(document).on("change keyup paste", "#amount", function() {
    $("#match_cents").val(parseInt($("#amount").val().replace(/,/g, "")) * 100);
  });
```